addons/purchase_stock/wizard/purchase_order_suggest.py

```diff
@@ -58,15 +59,22 @@ def _compute_estimated_price(self):
             seller_params = {'order_id': suggest.purchase_order_id}
 
             # Explicitly fetch existing records to avoid "NewId origin" shenanigans
-            # and add context's values needed by the monthly demand's compute.
-            products = self.env['product.product'].browse(self.product_ids.ids).with_context(
-                suggest._get_montlhy_demand_context()
-            )
+            products = self.env['product.product'].browse(self.product_ids.ids)
+
+            if suggest.based_on == 'actual_demand':
+                product_qty_dict = suggest._get_on_demand_products_qty()
+            else:
+                # and add context's values needed by the monthly demand's compute.
+                products = products.with_context(suggest._get_montlhy_demand_context())
+
```

Comment:
Based on the other comment, there should be no need for this branching.
```suggestion
products = self.env['product.product'].browse(self.product_ids.ids).with_context(
                suggest._get_montlhy_demand_context()
            )
```